packages/amplication-git-pull-request-service/src/core/commit/commits.service.ts

```diff
@@ -0,0 +1,189 @@
+import { GithubFactory, GitProvider } from '@amplication/git-service';
+import { Inject } from '@nestjs/common';
+import {
+  AmplicationLogger,
+  AMPLICATION_LOGGER_PROVIDER,
+} from '@amplication/nest-logger-module';
+
+const AMPLICATION = 'amplication';
+
+export class CommitsService {
+  constructor(
+    private githubBranchFactory: GithubFactory,
+    @Inject(AMPLICATION_LOGGER_PROVIDER)
+    private readonly logger: AmplicationLogger
+  ) {}
+
+  public async getBranch(
+    gitClient: GitProvider,
+    branch: string,
+    context: CommitContext
+  ): Promise<{ headCommit: string; defaultBranchName?: string }> {
+    this.logger.debug(`Getting branch ${branch}`, context);
+    let headCommit = (await gitClient.getBranch(branch))?.headCommit;
+    if (headCommit) {
+      this.logger.debug(
+        `Branch ${branch} head commit ${headCommit} was found`,
+        context
+      );
+      return {
+        headCommit,
+        defaultBranchName: undefined,
+      };
+    }
+    this.logger.warn(
+      `Branch ${branch} head commit was not found creating new branch`,
+      context
+    );
+
+    const defaultBranchName = await gitClient.getDefaultBranchName();
+    headCommit = (await gitClient.getBranch(defaultBranchName)).headCommit;
+    await gitClient.createBranch(branch, headCommit);
+    this.logger.info(`BRANCH ${branch} was created`, {
```

Comment:
To be consistent with other log messages, may I suggest `BRANCH` -> `Branch`, since the capitalized word has no functional meaning?